web/impact/impact/v1/views/office_hours_calendar_view.py

```diff
@@ -64,31 +78,107 @@ def _get_target_user(self, request):
+    def _office_hours_queryset(self):
+        if self.request_user_type == STAFF:
+            return self._staff_office_hours_queryset()
+        elif self.request_user_type == MENTOR:
+
+            return self._mentor_office_hours_queryset()
+        elif self.request_user_type == FINALIST:
+            return self._finalist_office_hours_queryset()
+        else:
+            self.response_elements['detail'] = self.NOT_OFFICE_HOURS_VIEWER
+            return self._null_office_hours_queryset()
+
+    def _staff_office_hours_queryset(self):
+        staff_programs = Clearance.objects.clearances_for_user(
+            self.target_user).values_list(
+                "program_family", flat=True)
+        in_visible_program_family = Q(
+            program_role__program__program_family__in=staff_programs)
+        program_mentors = ProgramRoleGrant.objects.filter(
+            ACTIVE_PROGRAM &
+            OFFICE_HOURS_HOLDER &
+            in_visible_program_family).values_list(
+                "person__id", flat=True)
+        active_mentors = Q(mentor__in=program_mentors)
+        return MentorProgramOfficeHour.objects.filter(
+            active_mentors,
+            start_date_time__range=[self.start_date, self.end_date]).order_by(
+                'start_date_time').annotate(
+                    finalist_count=Count("finalist"))
+
+    def _mentor_office_hours_queryset(self):
+        return MentorProgramOfficeHour.objects.filter(
              mentor=self.target_user,
-             start_date_time__range=[self.start_date, end_date]).order_by(
+             start_date_time__range=[self.start_date, self.end_date]).order_by(
+                 'start_date_time').annotate(
+                     finalist_count=Count("finalist"))
+
+    def _finalist_office_hours_queryset(self):
+        reserved_by_user = Q(finalist=self.target_user)
+        unreserved = Q(finalist__isnull=True)
+        user_programs = self.target_user.programrolegrant_set.filter(
+            ACTIVE_PROGRAM & OFFICE_HOURS_RESERVER).values_list(
+                "program_role__program_id", flat=True)
+        relevant_mentors = Q(**compose_filter(("mentor",
+                                               "programrolegrant",
+                                               "program_role",
+                                               "program",
+                                               "in"),
+                                              user_programs))
+        return MentorProgramOfficeHour.objects.filter(
+            reserved_by_user | unreserved & relevant_mentors,
+            start_date_time__range=[self.start_date, self.end_date])
+
+    def _null_office_hours_queryset(self):
+        return MentorProgramOfficeHour.objects.none()
+
+    def _get_office_hours_data(self):
+        primary_industry_key = "mentor__expertprofile__primary_industry__name"
+        office_hours = self._office_hours_queryset().filter(
+            program__isnull=True).order_by(
```

Comment:
Was this intentional? this currently filters out all the office hours that are associated with a program.

Reply:
Yes, I asked Jon to do this so office hours created on the old (current) calendar do not appear on the new calendar. They have different durations which causes some issues that didn't seem worth resolving - it seemed easier to maintain separation between the two interfaces.